### ISSUE TITLE
combine into `helpful_functions_repo`?

### ISSUE THREAD

**Issue by spinkney** [paul-buerkner/custom-brms-families#3]
Hi Paul,

I have a repo https://github.com/spinkney/helpful_stan_functions that is attempting to gather lots of stan function files written by users. I don't have a brms directory but it could be done. My goal is to have github actions that format the code for each PR so there's consistency. Also, instead of a big R file the stan functions would be in `.stanfunctions` files, which is a new file type that stanc3 supports and will be in 2.29.

Let me know if you think it would be worthwhile to merge into the helpful_stan_functions repo.

**Comment by paul-buerkner** [comment]
Thank you for this initiative! Can you explain what you think would be the benefits of moving it to your repo?

**Comment by spinkney** [comment]
Yes, users would have one consistent place to go to. It's possible this would become an official Stan repo, with some of the functions finding their way into Stan math. 

I hope that non R users could contribute which would benefit brms users by having a larger selection of Stan code to choose. 

I'm enforcing documentation about the function. It uses roxygen comments at the top like an R package. The documentation is updated automatically to a GitHub site that you can look at right now (link is at the repo). This helps to understand what the function is, what it does, and where to find more info.

I'm going to try to auto enforce code style. There's a merged pr in stanc3 with an auto formatter. I can run a bash script using GitHub actions over the '.stanfunctions' file to format and canonicalize (eg coverting the old array syntax to using the new 'array' keyword). This adds an element of consistency when using and perusing the code base.

There are some drawbacks to these. It's less of a free for all and takes time to maintain. Im not enforcing a brms example so users may still have to figure that out. But I think some users would contribute PRs for this. 

Martin brought up a point to consider in the forums. Of making this an R package that enhances brms. I think it's a good idea. I think my repo could be a GitHub submodule in such a package which houses all the Stan code. I don't know how cran deals with that.
